omnigibson/object_states/heat_source_or_sink.py

```diff
@@ -97,37 +93,27 @@ def get_optional_dependencies():
     def get_state_link_name():
         return m.HEATING_ELEMENT_LINK_NAME
 
-    def _compute_state_and_position(self):
-        # Find the link first. Note that the link is only required
-        # if the object is not in self.requires_inside mode.
-        heating_element_position = self.get_link_position()
-        if not self.requires_inside and heating_element_position is None:
-            return False, None
+    def _initialize(self):
+        # Run super first
+        super()._initialize()
+        self.initialize_link_mixin()
+
+    def _get_value(self):
+        # if the object is not in self.requires_inside mode, the link position is required.
+        if not self.requires_inside and self.get_link_position() is None:
+            return False
 
         # Check the toggle state.
         if self.requires_toggled_on and not self.obj.states[ToggledOn].get_value():
-            return False, None
+            return False
 
         # Check the open state.
         if self.requires_closed and self.obj.states[Open].get_value():
-            return False, None
-
-        # Return True and the heating element position (or None if not required).
-        return True, (heating_element_position if not self.requires_inside else None)
+            return False
 
-    def _initialize(self):
-        # Run super first
-        super()._initialize()
-        self.initialize_link_mixin()
-
-    def _update(self):
-        self.status, self.position = self._compute_state_and_position()
-
-    def _get_value(self):
-        return self.status, self.position
+        return True
```

Comment:
is the position not needed anymore because it's static (i.e.: it doesn't change?)

Reply:
We still need the position. It's just that we don't return position from the `get_value` call anymore. The code that uses the position will just do `state.get_link_position()`. I think this way we unify with other object states better (`HeatSourceOrSink` is the only state that returns two things from `get_value`).

Reply:
good point!